common/json_writer/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+use std::fmt::Write;
+
+#[derive(Debug, Clone)]
+pub struct JsonWriter {
+    buffer: String,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum JsonWriterError {
+    #[error("Failed to write the f64 value")]
+    Writef64ValueError(#[from] std::fmt::Error),
+}
```

Comment:
It's unclear to me how we can have such an error.

* According to doc of [`std::fmt`](https://doc.rust-lang.org/std/fmt/index.html#formatting-traits), string formatting is an infallible operation and the function `write_fmt` can fail only when the buffer fails, and that is *never* the case with  a [string buffer](https://doc.rust-lang.org/src/alloc/string.rs.html#2509-2521).
* Hence, I would prefer to encapsulate that weirdness inside this crate by 1) returning no error 2) unwraping the result of `write_fmt` with a comment.

@mneumann and @ChrisGreenaway would do you think of that?

Reply:
@didier-wenzek Well, think about NaN cases etc. for which we cannot create a valid JSON representation. That might be the only reason why we would need something like that.

If at all, it should be `WriteF64ValueError`.

Reply:
But this library puts "NaN" for the value, so doesn't fail. But of course, that's wrong, as you say, because it's not valid JSON. It's easy to be fast when you give up getting the right behaviour.

Reply:
This is the decisive argument. One needs to explicitly handle the NaN, Inf and -Inf cases in the `write_f64` method.